browser/extensions/api/brave_rewards_api.cc

```diff
@@ -282,45 +282,126 @@ ExtensionFunction::ResponseAction BraveRewardsTipSiteFunction::Run() {
   return RespondNow(NoArguments());
 }
 
-BraveRewardsTipTwitterUserFunction::BraveRewardsTipTwitterUserFunction()
+BraveRewardsTipUserFunction::BraveRewardsTipUserFunction()
     : weak_factory_(this) {
 }
 
-BraveRewardsTipTwitterUserFunction::~BraveRewardsTipTwitterUserFunction() {
+BraveRewardsTipUserFunction::~BraveRewardsTipUserFunction() {
 }
 
-ExtensionFunction::ResponseAction
-BraveRewardsTipTwitterUserFunction::Run() {
-  std::unique_ptr<brave_rewards::TipTwitterUser::Params> params(
-      brave_rewards::TipTwitterUser::Params::Create(*args_));
+ExtensionFunction::ResponseAction BraveRewardsTipUserFunction::Run() {
+  std::unique_ptr<brave_rewards::TipUser::Params> params(
+      brave_rewards::TipUser::Params::Create(*args_));
   EXTENSION_FUNCTION_VALIDATE(params.get());
 
   // Sanity check: don't allow tips in private / tor contexts,
   // although the command should not have been enabled in the first place.
   if (!brave::IsRegularProfile(browser_context())) {
-    return RespondNow(
-        Error("Cannot tip Twitter user in a private context"));
+    return RespondNow(Error("Cannot tip user in a private context"));
   }
 
   Profile* profile = Profile::FromBrowserContext(browser_context());
   auto* rewards_service = RewardsServiceFactory::GetForProfile(profile);
-  if (rewards_service) {
-    AddRef();
-    std::map<std::string, std::string> args;
-    args["user_id"] = params->media_meta_data.user_id;
-    args["twitter_name"] = params->media_meta_data.twitter_name;
-    args["screen_name"] = params->media_meta_data.screen_name;
-    rewards_service->SaveInlineMediaInfo(
-        params->media_meta_data.media_type,
-        args,
-        base::Bind(&BraveRewardsTipTwitterUserFunction::
-                   OnTwitterPublisherInfoSaved,
-                   weak_factory_.GetWeakPtr()));
+  if (!rewards_service) {
+    return RespondNow(Error("Rewards service is not initialized"));
   }
 
+  AddRef();
+
+  rewards_service->GetPublisherInfo(
+      params->publisher_key,
+      base::Bind(
+          &BraveRewardsTipUserFunction::OnTipUserGetPublisherInfo,
+          this));
+
   return RespondNow(NoArguments());
 }
 
+void BraveRewardsTipUserFunction::OnTipUserGetPublisherInfo(
+    const ledger::type::Result result,
+    ledger::type::PublisherInfoPtr info) {
+  if (result != ledger::type::Result::LEDGER_OK &&
+      result != ledger::type::Result::NOT_FOUND) {
+    Release();
+    return;
+  }
+
+  if (result == ledger::type::Result::NOT_FOUND) {
```

Comment:
can we do something like this 

```
if (result == ledger::type::Result::LEDGER_OK) {
  ShowTipDialog();
  Release();
 return;
}
```

and then logic in this if after that. This way we don't have this big if

Reply:
Fixed.